src/charm.py

```diff
@@ -1122,13 +1131,7 @@ def _handle_processes_failures(self) -> bool:
 
     def _set_primary_status_message(self) -> None:
         """Display 'Primary' in the unit status message if the current unit is the primary."""
-        try:
-            if self._patroni.get_primary(unit_name_pattern=True) == self.unit.name:
-                self.unit.status = ActiveStatus("Primary")
-            elif self._patroni.member_started:
-                self.unit.status = ActiveStatus()
-        except (RetryError, ConnectionError) as e:
-            logger.error(f"failed to get primary with error {e}")
+        self._set_active_status()
```

Comment:
We should most probably remove  this method and just call `_set_active_status()` directly.

Reply:
Removing it makes total sense. Thanks for catching that. It was removed on https://github.com/canonical/postgresql-k8s-operator/pull/435/commits/569046fe7cd028adda3001393829ac4ef54d7f03.